### DISCUSSION TITLE
Digital Ocean Mastodon Droplet Email Config. Answered.

### DISCUSSION THREAD

**Discussion by DavidGerhart** [mastodon/mastodon#21121] [Q&A] [answered]
A new Mastodon instance runs fine, but user registration confirmation emails never arrive.  What am I missing?

**Solution at the bottom**

The UBUNTU ufw settings:

Status: active
     To                         Action      From
     --                         ------      ----
[ 1] 22/tcp                     ALLOW IN    Anywhere                  
[ 2] 80/tcp                     ALLOW IN    Anywhere                  
[ 3] 443/tcp                    ALLOW IN    Anywhere                  
[ 4] 25/udp                     ALLOW IN    Anywhere                  
[ 5] 587/tcp                    ALLOW IN    Anywhere                  
[ 6] 22/tcp (v6)                ALLOW IN    Anywhere (v6)             
[ 7] 80/tcp (v6)                ALLOW IN    Anywhere (v6)             
[ 8] 443/tcp (v6)               ALLOW IN    Anywhere (v6)             
[ 9] 25/udp (v6)                ALLOW IN    Anywhere (v6)             
[10] 587 (v6)                   ALLOW IN    Anywhere (v6)             
[11] 587/tcp (v6)               ALLOW IN    Anywhere (v6)   

/home/mastodon/live .env.production file smtp lines:

SMTP_SERVER=smtp.mailgun.org
SMTP_PORT=587
SMTP_LOGIN=postmaster@mg.{fqdomian}
SMTP_PASSWORD={passphrase}
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_FROM_ADDRESS=Mastodon <notifications@{fqdomian}>

DNS records verified at MailGun (green check confirmed.)

CNAME | email.mg.{fqdomian}  | is an alias of mailgun.org
MX | mg.{fqdomian}  | mail handled by mxb.mailgun.org
MX | mg.{fqdomian} | mail handled by mxa.mailgun.org
TXT | mg.{fqdomian} | returns v=spf1 include:mailgun.org 
TXT | mx._domainkey.mg.{fqdomian} | returns k=rsa; p={passphrase)

What am I missing? Should the ufw port 587 be udp? 

Which error logs hold the clues?

Suggestions, improvements? Thanks much.

The setup code:

https://github.com/mastodon/mastodon/blob/main/lib/tasks/mastodon.rake#L302

Solution:

A single line in the /home/mastodon/live/.env.production file:

SMTP_OPENSSL_VERIFY_MODE=none  is incorrect for Mailgun
SMTP_OPENSSL_VERIFY_MODE=peer  is correct for Mailgun

**Comment by ghost** [comment]
Can you check sidekiq to see what went wrong with sending email?

**Reply by DavidGerhart** [reply]
> Can you check sidekiq to see what went wrong with sending email?

Thank you for asking!

Looks like SMTP Authentication Error 535... 

in 7 hours | 13 | default | UserMailer#welcome | {"_aj_globalid"=>"gid://mastodon/User/2"} | Net::SMTPAuthenticationError: 535 Authentication failed
-- | -- | -- | -- | -- | --
  | in 7 hours | 13 | default | UserMailer#confirmation_instructions | {"_aj_globalid"=>"gid://mastodon/User/2"}, "8yjrx4-NPmh7gKa1KCot" | Net::SMTPAuthenticationError: 535 Authentication failed

[in 7 hours](https://yankee.social/sidekiq/retries/1668915249.9471843-2c7b7312e085a185c3efe8e8)	13	[default](https://yankee.social/sidekiq/queues/default)	UserMailer#welcome	
{"_aj_globalid"=>"gid://mastodon/User/2"}
Net::SMTPAuthenticationError: 535 Authentication failed

[in 7 hours](https://yankee.social/sidekiq/retries/1668915542.9949017-687871d71fa134b7b184c1c0)	13	[default](https://yankee.social/sidekiq/queues/default)	UserMailer#confirmation_instructions	
{"_aj_globalid"=>"gid://mastodon/User/2"}, "8yjrx4-NPmh7gKa1KCot"
Net::SMTPAuthenticationError: 535 Authentication failed

**Reply by ghost** [reply]
Which MTA are you using?

**Reply by DavidGerhart** [reply]
mailgun

**Reply by DavidGerhart** [reply]
I'm due to run an errand, I'll be back in 45 minutes. She's waiting on me, and I'm burning domestic credits ...

**Reply by DavidGerhart** [reply]
back at my office now.

**Comment by DavidGerhart** [comment]
Thanks will do, report later...

David Gerhart
davidgerhart.com

**Comment by lookitmychicken** [comment]
How new is your DigitalOcean account? https://docs.digitalocean.com/support/why-is-smtp-blocked/

As non-techies, me and my fellow instance admin tried for 2 weeks to get our instance working via a DigitalOcean droplet. We logged a support ticket but it's over a week later and no response.

We gave up waiting and moved to a fully hosted service instead. :(

**Reply by ghost** [reply]
I'm not sure how th DigitalOcean thing work, but if you want to get an unmanaged server (which is cheaper) I'm wiling to setup Mastodon there for you.

**Reply by DavidGerhart** [reply]
Thanks for the offer... If I can't make the DO thing go ... I might talk to you. PM me about your setup and rates, tho?

**Reply by ghost** [reply]
I don't think I can PM you on Github. I have the instance called bassam.social.
You can contact me on Email (shown in my Mastodon instance), Twitter @LinuxArabia or telegram @AhmadDakhlallah

**Reply by DavidGerhart** [reply]
Thanks, I'll find your instance... you're up late! Sunday morning there already?

**Reply by ghost** [reply]
It'll be Sunday in 33 minutes

**Reply by jbold** [reply]
> How new is your DigitalOcean account? https://docs.digitalocean.com/support/why-is-smtp-blocked/
> 
> As non-techies, me and my fellow instance admin tried for 2 weeks to get our instance working via a DigitalOcean droplet. We logged a support ticket but it's over a week later and no response.
> 
> We gave up waiting and moved to a fully hosted service instead. :(

Which fully hosted service did you go with?

**Reply by lookitmychicken** [reply]
Toot.io https://toot.io/mastodon_hosting.html

They've been great ❤️

**Comment by DavidGerhart** [comment, marked as answer]
SMTP_OPENSSL_VERIFY_MODE=none changed to SMTP_OPENSSL_VERIFY_MODE=peer in the .env.production file for Mastodon
